app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java

```diff
@@ -0,0 +1,103 @@
+package com.github.bgabriel998.softwaredevproject;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.List;
+
+/**
+ * Adapter for gallery to show all images
+ */
+public class GalleryAdapter extends RecyclerView.Adapter<GalleryAdapter.ViewHolder> {
+
+    private Context mContext;
+    // TODO change to list of paths to images in folder
+    private List<Integer> images;
```

Comment:
images can be made final
```suggestion
    private final List<Integer> images;
```